src/main/java/net/sourceforge/argparse4j/impl/type/PathArgumentType.java

```diff
@@ -0,0 +1,371 @@
+                                      Path path) throws ArgumentParserException {
+        Path parent = path.getParent();
+        if (parent == null || !Files.isWritable(parent)) {
+            throw new ArgumentParserException(String.format(
+                    TextHelper.LOCALE_ROOT,
+                    "Cannot write parent of path: '%s'", path), parser, arg);
+        }
+    }
+
+    private void verifyCanCreate(ArgumentParser parser, Argument arg, Path path)
+            throws ArgumentParserException {
+        Path parent = path.getParent();
+        if (parent != null && Files.isWritable(parent)) {
+            return;
+        }
+        // An exception was thrown or the parent directory can't be written
+        throw new ArgumentParserException(String.format(TextHelper.LOCALE_ROOT,
+                "Cannot create file: '%s'", path), parser, arg);
+
+    }
+
+    private void verifyCanExecute(ArgumentParser parser, Argument arg, Path path)
+            throws ArgumentParserException {
+        if (!Files.isExecutable(path)) {
+            throw new ArgumentParserException(String.format(
+                    TextHelper.LOCALE_ROOT,
+                    "Insufficient permissions to execute file: '%s'", path),
+                    parser, arg);
+        }
+    }
+
+    private void verifyIsAbsolute(ArgumentParser parser, Argument arg, Path path)
+            throws ArgumentParserException {
+        if (!path.isAbsolute()) {
+            throw new ArgumentParserException(
+                    String.format(TextHelper.LOCALE_ROOT,
+                            "Not an absolute file: '%s'", path), parser, arg);
+        }
+    }
+
+    private void verifyMinDepth(ArgumentParser parser, Argument arg, Path path)
+            throws ArgumentParserException {
+        if (path.getNameCount() < pathMinDepth) {
+            throw new ArgumentParserException(
+                    String.format(TextHelper.LOCALE_ROOT,
+                            "Does not meet depth=%d requirement for path: '%s' ", pathMinDepth, path), parser, arg);
+        }
+    }
+
+    private void verifyIsEmpty(ArgumentParser parser, Argument arg, Path path)
+            throws ArgumentParserException {
+        try {
+            if (Files.isDirectory(path) && Files.newDirectoryStream(path).iterator().hasNext()) {
+                throw new ArgumentParserException(String.format(
+                        TextHelper.LOCALE_ROOT, "Path is not empty: '%s' ", path),
+                        parser, arg);
+            }
+            if (Files.size(path) > 0) {
+                throw new ArgumentParserException(String.format(
+                        TextHelper.LOCALE_ROOT, "File is not empty: '%s' ", path),
+                        parser, arg);
+            }
+        } catch (IOException e) {
+            throw new ArgumentParserException(String.format(
+                    TextHelper.LOCALE_ROOT, "%s", e.getMessage()),

```

Comment:
It would be better to describe the situation that the error occurred while verifying emptiness of the path, instead of just formatting IOException message.